### PR TITLE
stm32: add support for I2C4

### PR DESCRIPTION
**Pull request by vaussard**
Add Kconfig/code/dts/fixup/pinmux to support I2C4, as found on stm32l496 for example.

### CODE REVIEW

File: drivers/i2c/i2c_ll_stm32.c [hunk 1]
```diff
@@ -313,4 +313,47 @@ static void i2c_stm32_irq_config_func_3(struct device *dev)
...
+#endif /* I2C4_BASE */
+
+#ifdef CONFIG_I2C_STM32_INTERRUPT
+static void i2c_stm32_irq_config_func_4(struct device *port);
+#endif
+
+static const struct i2c_stm32_config i2c_stm32_cfg_4 = {
+	.i2c = (I2C_TypeDef *)CONFIG_I2C_4_BASE_ADDRESS,
+	.pclken = {
+		.enr = LL_APB1_GRP2_PERIPH_I2C4,
```
Comment (ydamigos) [inline drivers/i2c/i2c_ll_stm32.c]: On STM32F7 series it is LL_APB1_GRP1_PERIPH_I2C4. Could you add an #ifdef here?

**Review by ydamigos — CHANGES_REQUESTED** [review]
LGTM. Just a comment.

**Review by erwango — APPROVED** [review]

File: drivers/i2c/i2c_ll_stm32.c (hunk 1 above)
Comment (erwango) [reply drivers/i2c/i2c_ll_stm32.c]: F7 is not yet ported. I prefer to keep it separate.
Even if it should arrive rather soon, this PR might be merged in 1.12 while F7 will likely be part of 1.13

**Review by ydamigos — APPROVED** [review]

File: drivers/i2c/i2c_ll_stm32.c (hunk 1 above)
Comment (ydamigos) [reply drivers/i2c/i2c_ll_stm32.c]: To make myself clear clear, I was talking about the following #ifdef which allows the PR to be merged in 1.12:
```
#ifdef SOC_SERIES_STM32L4X
         .enr = LL_APB1_GRP2_PERIPH_I2C4,
         .bus = STM32_CLOCK_BUS_APB1_2,
#else
         .enr = LL_APB1_GRP1_PERIPH_I2C4,
         .bus = STM32_CLOCK_BUS_APB1,
#endif
```

**Pull request merged**